src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1297,8 +1329,12 @@ private static void ReportAnyMismatchedConstraints(MethodSymbol interfaceMethod,
                         return;
                     }
 
-                    //if we haven't found a match, do a weaker comparison that ignores static-ness, accessibility, and return type
-                    if ((object)closeMismatch == null && implementingTypeIsFromSomeCompilation)
+                    // If we haven't found a match, do a weaker comparison that ignores static-ness, accessibility, and return type.
+                    // But do this only if interface member is public because language doesn't allow implicit implementations for
```

Comment:
How does  "But do this only if interface member is public because language doesn't allow implicit implementations for **non-public** members ..." sound?